lib/widgets/resources/arm.dart

```diff
@@ -0,0 +1,280 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:viam_sdk/viam_sdk.dart';
+import 'package:viam_sdk/widgets.dart';
+
+/// A widget to control an [Arm].
+class ViamArmWidget extends StatefulWidget {
+  /// The [Arm]
+  final Arm arm;
+
+  const ViamArmWidget({
+    Key? key,
+    required this.arm,
+  }) : super(key: key);
+
+  @override
+  State<ViamArmWidget> createState() => _ViamArmWidgetState();
+}
+
+class _ViamArmWidgetState extends State<ViamArmWidget> {
+  Pose endPosition = Pose();
+  List<double> jointPositions = [];
+
+  Future<void> _positions() async {
+    final ep = await widget.arm.endPosition();
+    final jp = await widget.arm.jointPositions();
+    setState(() {
+      jointPositions = jp;
+      endPosition = ep;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _positions();
+  }
+
+  Future<void> updateEndPosition(String field, double increment) async {
+    final ep = endPosition;
+    if (field == 'x') {
+      ep.x += increment;
+    } else if (field == 'y') {
+      ep.y += increment;
+    } else if (field == 'z') {
+      ep.z += increment;
+    } else if (field == 'theta') {
+      ep.theta += increment;
+    } else if (field == 'ox') {
+      ep.oX += increment;
+    } else if (field == 'oy') {
+      ep.oY += increment;
+    } else if (field == 'oz') {
+      ep.oZ += increment;
+    }
+    await widget.arm.moveToPosition(ep);
+    await _positions();
+  }
```

Comment:
[suggestion] I think the string comparison here is _fine_, I might suggest defining an enum for the `fields` with the different options, and using a switch statement instead.

Not something blocking merging, just a preference on my side.